src/containerapp/azext_containerapp/_clients.py

```diff
@@ -1009,6 +1011,40 @@ def create(cls, cmd, resource_group_name, name, session_pool_envelope, no_wait=F
 
         r = send_raw_request(cmd.cli_ctx, "PUT", request_url, body=json.dumps(session_pool_envelope))
 
+        # try to add user as session pool creator role to the session pool
+        try:
+            # get princpalId of the user
+            principal_id_url = "https://graph.microsoft.com/v1.0/me"
+            principal_id = send_raw_request(cmd.cli_ctx,"GET",principal_id_url).json()['id']
+            scope = "subscriptions/{}/resourceGroups/{}/providers/Microsoft.App/sessionPools/{}".format(sub_id, resource_group_name, name)
+            role_assignment_fmt = "{}/{}/providers/Microsoft.Authorization/roleAssignments/{}?api-version=2022-04-01"
+            role_assignment_url = role_assignment_fmt.format(
+                management_hostname.strip('/'),
+                scope,
+                uuid.uuid4()
+            )
+            role_definition_id = "/{}/providers/Microsoft.Authorization/roleDefinitions/{}".format(
+                scope,
+                SESSION_CREATOR_ROLE_ID)
+            assign_role_r = send_raw_request(cmd.cli_ctx,"PUT", role_assignment_url, body=json.dumps({
+                "properties": {
+                    "roleDefinitionId": role_definition_id,
+                    "principalId": principal_id
+                }
+            }))
+        # if anything goes wrong print error but do not throw error
```

Comment:
These logic are for do the role assignment, please move them to the decorator, don't set in the `SessionPoolPreviewClient.create`.

Reply:
switched logic to decorator